pkg/tags/tags.go

```diff
@@ -53,17 +53,29 @@ type Tags struct {
 
 // NewTags creates a tags object
 func NewTags(stateStore storage.StateStorer, logger logging.Logger) *Tags {
+	rand.Seed(time.Now().Unix())
```

Comment:
```suggestion
	rand.Seed(time.Now().UnixNano())
```
After this, you should be able to remove the sleep from the test as well.

Reply:
Well, the sleep is still needed for some reason, though now only for a nanosecond